sql/src/main/java/io/crate/analyze/relations/AnalyzedRelation.java

```diff
@@ -43,8 +43,6 @@
 
     Field getField(ColumnIdent path, Operation operation) throws UnsupportedOperationException, ColumnUnknownException;
 
-    List<Field> fields();
```

Comment:
Maybe we actually keep it here with an added `Nonull`  annotation, since a statement that implements `AnalyzedRelation` will have `fields`.

Reply:
I added `NonNull` to `AnalyzedRelation` but since annotations are inherited, also to the relevant subclasses. I hope this makes sense to you.